src/result_processor.c

```diff
@@ -905,6 +917,19 @@ static int ReturnResult(SearchResult *buffered_result,  SearchResult *result_out
   return RS_RESULT_OK;
 }
 
+static int LockGil(RPBufferAndLocker *rpBufferAndLocker, RedisModuleCtx* redisCtx) {
```

Comment:
`LockGIL` or LockRedis in some way. GIL is a redis term so I'm not sure if we want to use it